### ISSUE TITLE
azure-cli 2.24.0 writes to STDERR

### ISSUE THREAD

**Issue by msftradford** [Azure/azure-cli#18262]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
We have been deploying to MS-hosted agents without issue. It looks like some of the agents started picking up azure-cli v2.24.0 recently and our deployments now fail with: `2021-05-26T21:13:12.9875423Z ##[error]Script failed with error: Error: The process 'D:\a\_temp\azureclitaskscript1622063412668.bat' failed because one or more lines were written to the STDERR stream
2021-05-26T21:13:12.9964337Z [command]C:\Windows\system32\cmd.exe /D /S /C ""C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az.cmd" account clear"` The lines immediately before the error statement are the logs from a call to `az container create` There are no other errors reported in the tasklog. This error does not occur when we are assigned an agent with azure-cli v2.23.0. The error is not fixed by unchecking "fail on standard error" in the Azure CLI task.

**To Reproduce**
Be assigned an agent with v2.24.0 and run the Azure CLI task to create a container

**Expected behavior**
The Azure CLI task should succeed on both v2.23.0 and v2.24.0

**Environment summary**
Microsoft-hosted agent w/ vs2017-win2016 image
Azure pipelines Azure CLI task v1.164.0
azure-cli 2.24.0
core 2.24.0
telemetry 1.0.6
azure-devops 0.18.0

**Comment by msftradford** [comment]
I ended up piping the errors to a file and this definitely happens on a call to `az container create`. I'm attaching the output of the error pipe and it looks like just a single empty line logged to STDERR
[container_create - Example.log](https://github.com/Azure/azure-cli/files/6556662/container_create.-.Example.log)

**Comment by msftradford** [comment]
Reverted my local version to azure-cli 2.23 and re-ran the exact same command and nothing was written to STDERR. Cannot attach the log file produced by the STDERR pipe on the container create command because it is just an empty file. This is different than v2.24 where there is a single empty line in the file.

**Comment by msftradford** [comment]
Enabled --debug on the create command and did the same test with the two versions. Interestingly enough, there is an extra empty line in the error log for v2.24 that is not in v2.23. The line appears after the response content from a PUT with request url: 'https://management.azure.com/subscriptions/{SUBSCRIPTION_ID}/resourceGroups/{RESOURCE_GROUP}/providers/Microsoft.ContainerInstance/containerGroups/{CONTAINER_INSTANCE}?api-version=2018-10-01' and immediately before these lines: 

```
DEBUG: cli.knack.cli: Event: CommandInvoker.OnTransformResult [<function _resource_group_transform at 0x04357B20>, <function _x509_from_base64_to_hex_transform at 0x04357B68>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnFilterResult []
DEBUG: cli.knack.cli: Event: Cli.SuccessfulExecute []
DEBUG: cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x04321190>]
INFO: az_command_data_logger: exit code: 0
INFO: cli.__main__: Command ran in 2.919 seconds (init: 1.203, invoke: 1.715)
INFO: telemetry.save: Save telemetry record of length 3092 in cache
```

**Comment by yonzhan** [comment]
route to service team

**Comment by jiasli** [comment]
@msftradford, the log file you shared is empty: https://github.com/Azure/azure-cli/files/6556807/container_create.-.v2.24.Example.log

Could you share the actual content in STDERR? Is it the same as #18276?

**Comment by msftradford** [comment]
hi @jiasli! that's right. It appears to be a completely empty file but it actually has one single line composed of 70 space characters and I believe that single line is causing the task to fail (only with v2.24). This is different from the actually totally empty file that was created with v2.23.0. The issue that you mentioned appears unrelated - we don't receive any useful error messages beyond the single "empty" line with 70 space characters. This could still be an incompatibility issue between azure-cli and Python, but I'm not sure if it is the exact same issue.

**Comment by jiasli** [comment]
@msftradford, I probably needs to route this issue to service team to investigate... Could you repro this issue locally?

**Comment by msftradford** [comment]
yes! the piping to log files was done by running the commands locally, that is when I noticed the empty line in the STDERR pipe.

**Comment by jiasli** [comment]
I am able to reproduce with 

```
az container create --resource-group myResourceGroup --name mycontainer2 --image mcr.microsoft.com/azuredocs/aci-helloworld --dns-name-label aci-demo060702 --ports 80 2>stderr.txt
```

Looks like it has something to do with progress bar. 

![image](https://user-images.githubusercontent.com/4003950/120952136-0134a980-c77d-11eb-89b7-b4e10bae2e0b.png)

\+ @Juliehzl to help take a look.
